app/models/user.rb

```diff
@@ -591,6 +595,10 @@ def cohorts
     portal_teacher ? portal_teacher.cohorts : (portal_student ? portal_student.cohorts : [])
   end
 
+  def cohort_projects
+    cohorts.map {|c| c.project}.flatten.compact.uniq
```

Comment:
This might be better defined like this: `has_many :cohort_projects, through: :cohorts, source: :project, uniq: true`
I'm not sure I've got that right. But by doing it that it should be a bit more efficient because it ought to use a SQL to do the map/join, and uniq.